From 84da59c0891ce82bddd02281d8647e3af2963f32 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 16 Jan 2023 17:03:24 +0100
Subject: [PATCH 0122/1695] plat-stm32mp1: conf: default disable
 CFG_EXTERNAL_DT for STM32MP15

Upstream U-boot and Linux device tree alignment constraint does not
apply for downstream OP-TEE.

This reverts releases commit 2555fbce8a1333457c1f28e9b5067aa11c7ee74a
constraint.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 164059d576ae66c95bf6e75933b9aab456f6f399 plat-stm32mp1: conf: default disable CFG_EXTERNAL_DT for STM32MP15]
Change-Id: Icdd971ace115e331e396f70ca24a5d0dd0e53477
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index ba454c32b..9929cc787 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -86,6 +86,9 @@ ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP15)),)
 $(call force,CFG_STM32MP15,y)
 endif
 
+# Default do not access external DT passed to non-secure boot stage
+CFG_EXTERNAL_DT ?= n
+
 # CFG_STM32MP1x switches are exclusive.
 # - CFG_STM32MP15 is enabled for STM32MP15x-* targets (default)
 # - CFG_STM32MP13 is enabled for STM32MP13x-* targets
@@ -133,7 +136,6 @@ $(call force,CFG_STM32MP13_CLK,y)
 $(call force,CFG_STM32MP13_REGULATOR_IOD,y)
 $(call force,CFG_TEE_CORE_NB_CORE,1)
 $(call force,CFG_WITH_NSEC_GPIOS,n)
-CFG_EXTERNAL_DT ?= n
 CFG_STM32MP_OPP_COUNT ?= 2
 CFG_WITH_PAGER ?= n
 endif # CFG_STM32MP13
@@ -148,7 +150,6 @@ $(call force,CFG_STM32_SAES,n)
 $(call force,CFG_STM32MP15_CLK,y)
 CFG_CORE_RESERVED_SHM ?= n
 CFG_HALT_CORES_ON_PANIC ?= y
-CFG_EXTERNAL_DT ?= y
 CFG_STM32_BSEC_SIP ?= y
 CFG_TEE_CORE_NB_CORE ?= 2
 CFG_WITH_SOFTWARE_PRNG ?= y
-- 
2.47.2

